main.go

```diff
@@ -404,7 +407,22 @@ func main() {
 
 		}
 
-		if err = otelv1beta1.SetupCollectorWebhook(mgr, cfg, reviewer, crdMetrics); err != nil {
+		bv := func(collector otelv1beta1.OpenTelemetryCollector) admission.Warnings {
```

Comment:
Maybe we should put this method on the collector controller such that we can just pass a reference to its function rather than inlining this closure.

Reply:
I agree